ams/server/src/main/java/com/netease/arctic/server/table/executor/BaseTableExecutor.java

```diff
@@ -62,25 +67,36 @@ protected void initHandler(List<TableRuntimeMeta> tableRuntimeMetaList) {
         .map(tableRuntimeMeta -> tableRuntimeMeta.getTableRuntime())
         .filter(tableRuntime -> enabled(tableRuntime))
         .forEach(
-            tableRuntime ->
-                executor.schedule(
-                    () -> executeTask(tableRuntime), getStartDelay(), TimeUnit.MILLISECONDS));
+            tableRuntime -> {
+              ScheduledFuture<?> scheduledFuture =
+                  executor.scheduleWithFixedDelay(
+                      () -> executeTask(tableRuntime),
+                      getStartDelay(),
+                      getNextExecutingTime(tableRuntime),
+                      TimeUnit.MILLISECONDS);
+              scheduledTasks.put(tableRuntime.getTableIdentifier(), scheduledFuture);
+            });
     logger.info("Table executor {} initialized", getClass().getSimpleName());
   }
 
   private void executeTask(TableRuntime tableRuntime) {
     if (isExecutable(tableRuntime)) {
-      try {
-        execute(tableRuntime);
-      } finally {
-        scheduleIfNecessary(tableRuntime, getNextExecutingTime(tableRuntime));
-      }
+      execute(tableRuntime);
     }
   }
 
   protected final void scheduleIfNecessary(TableRuntime tableRuntime, long millisecondsTime) {
     if (isExecutable(tableRuntime)) {
-      executor.schedule(() -> executeTask(tableRuntime), millisecondsTime, TimeUnit.MILLISECONDS);
+      if (scheduledTasks.containsKey(tableRuntime.getTableIdentifier())) {
```

Comment:
What I originally thought was that I should cancel the previous task, because I think the original intention of handleConfigChanged is that if the configuration changes, the task should be scheduled according to the new configuration. But from the perspective of solving this issue, it would be better to be simpler.